schemas/annotations.yaml

```diff
@@ -0,0 +1,48 @@
+openapi: '3.0.2'
+info:
+  title: Annotations Example
+  version: '1.0'
+
+components:
+  schemas:
+    Dog:
+      type: object
+      properties:
+        bark:
+          type: boolean
+        breed:
+          type: string
+          minLength: 1
+          maxLength: 50
+        name:
+          type: string
+          minimum: 0
+          maximum: 10
+        hairColor:
+          type: string
+          pattern: "^[a-zA-Z-0-9 ]{3,100}|null$"
+        eyeColor:
+          type: string
+          pattern: "^[a-zA-Z-0-9 ]{3,100}|null$"
+          minLength: 1
+          maxLength: 50
+          minimum: 0
+          maximum: 10
```

Comment:
`|null` is confusing here - the `hairColor` should be able to be omitted, and `null` would still match the other half of the pattern.

Also, what would the competing minimum/minLength do here? I feel like these are invalid annotation combinations for testing.

Reply:
Schema updated